pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -27,19 +29,18 @@ import (
 )
 
 func newETCD3Storage(c storagebackend.Config) (storage.Interface, DestroyFunc, error) {
-	tlsInfo := transport.TLSInfo{
-		CertFile: c.CertFile,
-		KeyFile:  c.KeyFile,
-		CAFile:   c.CAFile,
-	}
-	tlsConfig, err := tlsInfo.ClientConfig()
-	if err != nil {
-		return nil, nil, err
-	}
-	// NOTE: Client relies on nil tlsConfig
-	// for non-secure connections, update the implicit variable
-	if len(c.CertFile) == 0 && len(c.KeyFile) == 0 && len(c.CAFile) == 0 {
-		tlsConfig = nil
+	var tlsConfig *tls.Config
+	if !(len(c.CertFile) == 0 && len(c.KeyFile) == 0 && len(c.CAFile) == 0) {
```

Comment:
It also seems weird in previous code path. If it failed, it would return error. But if it didn't and some of this params is empty, it will set tlsConfig to nil again..